tests/publiccloud/aistack_basic.pm

```diff
@@ -0,0 +1,136 @@
+    my $user = $testapi::username;
+    my $email = "you\@example.com";
+    my $helm_values = get_var('HELM_CONFIG', 'https://github.com/yarunachalam/os-autoinst-distri-opensuse/tree/aistack_basic/data/aistack_values.yaml');
+    my $repo_url = 'https://github.com/SUSE/private-ai-charts.git';
+
+
+    set_var('APPLICATION_COLLECTION_USER_PASSWORD_SECRET', '<application_collection_user_token>');
+
+
+    # config kubectl
+    assert_script_run("mkdir -p ~/.kube");
+    assert_script_run("sudo cp /etc/rancher/rke2/rke2.yaml ~/.kube/config");
+    assert_script_run("kubectl config get-contexts");
+    assert_script_run("kubectl config use-context default");
+    assert_script_run("kubectl config view");
+
+    # Add cert-manager repo,install
+    assert_script_run("helm repo add $cert_repo");
+    assert_script_run("helm repo update");
+    assert_script_run("helm install cert-manager jetstack/cert-manager --namespace cert-manger --create-namespace --version v1.15.2 --set crds.enabled=true");
+
+    sleep 180;
+    assert_script_run("helm list --all-namespaces");
+    assert_script_run("kubectl get pods --all-namespaces");
+
+    # Access to Application collection registery
+    # Get docker username and password
+    assert_script_run("kubectl create ns suse-private-ai");
+    assert_script_run("kubectl create secret docker-registry application-collection --docker-server=dp.apps.rancher.io --docker-username='$docker_user_name' --docker-password='$APPLICATION_COLLECTION_USER_PASSWORD_SECRET' -n suse-private-ai", timeout => 360);
+    # Add suse-private-ai install
+    #    assert_script_run("sudo -u $user mkdir -p /home/$user/.ssh");
+    #assert_script_run("ssh-keyscan -H github.com >> /home/$user/.ssh/known_hosts", timeout => 60);
+    #assert_script_run('eval $(ssh-agent -s)');
+    #assert_script_run("sudo -u $user ssh-keygen -t rsa -N \"\" -f /home/$user/.ssh/id_rsa");
+    #assert_script_run("sudo -u $user ssh-add /home/$user/.ssh/id_rsa");
+    #assert_script_run("sudo chmod 700 /home/$user/.ssh && sudo chmod 600 /home/$user/.ssh/id_rsa && sudo chmod 644 /root/.ssh/id_rsa.pub");
+    #assert_script_run("ls -l /home/$user/.ssh && ls -l /home/$user/.ssh/id_rsa && ls -l /root/.ssh/id_rsa.pub && ls -la /home/$user/.ssh");
+    # assert_script_run("git init");
+    # assert_script_run("git config --global http.sslVerify false");
+    # assert_script_run('git config --global credential.helper cache --timeout=3600');
+    # assert_script_run("git config --global user.email \"$email\"");
+    # assert_script_run("git config --global user.name \"$user\"");
+    # assert_script_run('git config --global credential.helper cache');
+     assert_script_run("git clone '$repo_url'");
+     assert_script_run("curl -sSL -O $helm_values");
+     assert_script_run("helm upgrade --install suse-private-ai private-ai-charts/Chart.yaml --namespace suse-private-ai --create-namespace --values aistack_values.yaml --debug");
+     sleep 180;
```

Comment:
Is this sleep to give time to k3s to start? if so, maybe this is better:

```
assert_script_run("command |& tee k3s-private-ai.log | grep \"I KNOW THIS LINE TELLS ME MY NODE IS READY\"", timeout => 180, fail_message => "No self healing");
```

So the assert script run checks for you, and you can add the logfile to be uploaded in the post fail hook.

Reply:
can you refer some existing code for this

Reply:
@yarunachalam you can see for instance: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20644/files#diff-d2abfd176e31532049f3e9fc2d51db12a56f87fb7b714939e42b3ee703df9016R49

(similar approach, more reliable than the sleep)